src/handler/http/request/dashboards/mod.rs

```diff
@@ -99,80 +105,41 @@ pub async fn create_dashboard(
         ("dashboard_id" = String, Path, description = "Dashboard ID"),
     ),
     request_body(
-        content = Dashboard,
+        content = UpdateDashboardRequestBody,
         description = "Dashboard details",
     ),
     responses(
-        (status = StatusCode::OK, description = "Dashboard updated", body = HttpResponse),
+        (status = StatusCode::OK, description = "Dashboard updated", body = UpdateDashboardResponseBody),
         (status = StatusCode::NOT_FOUND, description = "Dashboard not found", body = HttpResponse),
         (status = StatusCode::INTERNAL_SERVER_ERROR, description = "Failed to update the dashboard", body = HttpResponse),
     ),
 )]
 #[put("/{org_id}/dashboards/{dashboard_id}")]
 async fn update_dashboard(
     path: web::Path<(String, String)>,
-    body: web::Bytes,
+    req_body: web::Json<UpdateDashboardRequestBody>,
     req: HttpRequest,
 ) -> impl Responder {
     let (org_id, dashboard_id) = path.into_inner();
     let query = web::Query::<HashMap<String, String>>::from_query(req.query_string()).unwrap();
     let folder = crate::common::utils::http::get_folder(&query);
     let hash = query.get("hash").map(|h| h.as_str());
-    let Ok(dashboard) = parse_dashboard(body) else {
-        return MetaHttpResponse::internal_error("Error parsing dashboard");
-    };
-    match dashboards::update_dashboard(&org_id, &dashboard_id, &folder, dashboard, hash).await {
-        Ok(dashboard) => HttpResponse::Ok().json(dashboard),
-        Err(err) => err.into(),
-    }
-}
-
-/// HTTP URL query component that contains parameters for listing dashboards.
-#[derive(Debug, Deserialize, utoipa::IntoParams)]
-#[into_params(style = Form, parameter_in = Query)]
-pub struct ListQuery {
-    /// Optional folder ID filter parameter
-    ///
-    /// If neither `folder` nor any other filter parameter are set then this
-    /// will search for all dashboards in the "default" folder.
-    ///
-    /// If `folder` is not set and another filter parameter, such as `title`, is
-    /// set then this will search for dashboards in all folders.
-    folder: Option<String>,
 
-    /// The optional case-insensitive title substring with which to filter
-    /// dashboards.
-    title: Option<String>,
-}
-
-impl ListQuery {
```

Comment:
This is moved into `http/models` along with the other structs that define the HTTP API schemas